packages/smithy-client/src/date-utils.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { parseEpochTimestamp, parseRfc3339DateTime, parseRfc7231DateTime } from "./date-utils";
+
+describe("parseRfc3339DateTime", () => {
+  it.each([null, undefined])("returns undefined for %s", (value) => {
+    expect(parseRfc3339DateTime(value)).toBeUndefined();
+  });
+
+  describe("parses properly formatted dates", () => {
+    it("with fractional seconds", () => {
+      expect(parseRfc3339DateTime("1985-04-12T23:20:50.52Z")).toEqual(new Date(Date.UTC(1985, 3, 12, 23, 20, 50, 520)));
+    });
+    it("without fractional seconds", () => {
+      expect(parseRfc3339DateTime("1985-04-12T23:20:50Z")).toEqual(new Date(Date.UTC(1985, 3, 12, 23, 20, 50, 0)));
+    });
+    it("with leap seconds", () => {
+      expect(parseRfc3339DateTime("1990-12-31T15:59:60Z")).toEqual(new Date(Date.UTC(1990, 11, 31, 15, 59, 60, 0)));
+    });
+    it("with leap days", () => {
+      expect(parseRfc3339DateTime("2004-02-29T15:59:59Z")).toEqual(new Date(Date.UTC(2004, 1, 29, 15, 59, 59, 0)));
+    });
+  });
+
+  it.each([
```

Comment:
You should have tests in all of these where there's some extra leading or trailing garbage to make sure the whole string is matched.

Reply:
Sure. It is; I don't know why anchors didn't work for you previously but they do for me, I verified after our dicussion yesterday.

Reply:
Teach me your ways magic man